src/Bundle/Resources/config/services/state/provider.xml

```diff
@@ -18,30 +18,39 @@
-        <service id="sylius.state_provider.factory" class="Sylius\Resource\State\Provider\FactoryProvider" decorates="sylius.state_provider.read" decoration-priority="500">
+        <service id="sylius.state_provider.factory" class="Sylius\Resource\State\Provider\FactoryProvider" decorates="sylius.state_provider.read" decoration-priority="600">
             <argument type="service" id=".inner" />
             <argument type="service" id="sylius.state_factory" />
         </service>
 
-        <service id="sylius.state_provider.dispatch_post_read_event" class="Sylius\Resource\Symfony\EventDispatcher\State\DispatchPostReadEventProvider" decorates="sylius.state_provider.read" decoration-priority="400"
+        <service id="sylius.state_provider.dispatch_post_read_event" class="Sylius\Resource\Symfony\EventDispatcher\State\DispatchPostReadEventProvider" decorates="sylius.state_provider.read" decoration-priority="500"
         >
             <argument type="service" id=".inner" />
             <argument type="service" id="sylius.dispatcher.operation" />
         </service>
 
-        <service id="sylius.state_provider.deserialize" class="Sylius\Resource\Symfony\Serializer\State\DeserializeProvider" decorates="sylius.state_provider.read" decoration-priority="300">
+        <service id="sylius.state_provider.deserialize" class="Sylius\Resource\Symfony\Serializer\State\DeserializeProvider" decorates="sylius.state_provider.read" decoration-priority="400">
             <argument type="service" id=".inner" />
             <argument type="service" id="serializer" on-invalid="null" />
         </service>
 
-        <service id="sylius.state_provider.form" class="Sylius\Resource\Symfony\Form\State\FormProvider" decorates="sylius.state_provider.read" decoration-priority="200">
+        <service id="sylius.state_provider.form" class="Sylius\Resource\Symfony\Form\State\FormProvider" decorates="sylius.state_provider.read" decoration-priority="300">
             <argument type="service" id=".inner" />
             <argument type="service" id="sylius.form.factory" />
         </service>
 
-        <service id="sylius.state_provider.validate" class="Sylius\Resource\Symfony\Validator\State\ValidateProvider" decorates="sylius.state_provider.read" decoration-priority="100">
+        <service id="sylius.state_provider.validate" class="Sylius\Resource\Symfony\Validator\State\ValidateProvider" decorates="sylius.state_provider.read" decoration-priority="200">
             <argument type="service" id=".inner" />
             <argument type="service" id="validator" />
         </service>
+
+        <service id="sylius.state_provider.security" class="Sylius\Resource\State\Provider\SecurityProvider" decorates="sylius.state_provider.read" decoration-priority="100">
+            <argument type="service" id=".inner" />
+            <argument type="service" id="sylius.state_provider.security_attribute" />
+            <argument type="service" id="security.authorization_checker" />
```

Comment:
hum... Is this in optional requirements? in this case, you shoud handle this with NullOnInvalid

Reply:
I will rewrite this part to copy the behavior of APIP : https://github.com/api-platform/core/blob/main/src/State/Provider/SecurityParameterProvider.php#L53

Reply:
yes, but I think they removed the service when symfony/security is not available. There are two solution, remove it or set null if the service is invalid an throw an error to "try to run composer require ..."

Reply:
@loic425 you mean this : https://github.com/api-platform/core/blob/main/src/Symfony/Bundle/DependencyInjection/ApiPlatformExtension.php#L838-L849 ?